src/app/app.ts

```diff
@@ -0,0 +1,56 @@
+import {Post} from 'mattermost-redux/types/posts';
+import {AppCallResponse} from 'mattermost-redux/types/apps';
+import {Tickets} from 'node-zendesk';
+
+import mmClient from '../mattermost/client';
+import zendeskClient from '../zendesk/client';
+
+import store from './store';
+
+const username = process.env.ZENDESK_USERNAME as string;
+const token = process.env.ZENDESK_API_TOKEN as string;
+const apiURL = process.env.ZENDESK_URL + '/api/v2' as string;
+
+class App {
+    async createTicketFromPost(ticket, channelId: string, userId: string, postId: string) {
+        const zdClient = zendeskClient(username, token, apiURL);
+        const result = await zdClient.tickets.create(ticket);
+        const user = await zdClient.users.show(result.requester_id);
+        const host = process.env.ZENDESK_URL;
+
+        const message = `${user.name} created ticket [#${result.id}](${host}/agent/tickets/${result.id}) [${result.subject}]`;
+
+        const post: Post = {
+            message,
+            channel_id: channelId,
+            root_id: postId,
+        };
+
+        const client = mmClient;
+        client.setToken(store.getBotAccessToken());
+        const pRes = client.createPost(post);
```

Comment:
I looked into using the MD response field and it does work, but posts an ephemeral message from system. In this case, I want the Zendesk bot to post a thread response to the root post letting others in Mattermost know that the post was created as a Zendesk ticket.  

I haven't checked extensively to see if this is available, but didn't see anything to accomplish this without using the MM Client4 createPost API .

![image](https://user-images.githubusercontent.com/7575921/100301603-17f07d00-2f5e-11eb-9b43-0654786c06e6.png)

Reply:
You are right. This usecase should be done with your approach.

Nevertheless, @levb @mickmister should we try to ease this kind of usecase? I have mixed feelings, because the bot may or may not have permission to write in that channel, so it would be "dangerous" to write directly from `plugin-apps` the post.

And on a different note, instead of a System message, should the ephemeral be a message from the app bot?

Reply:
Ticket for the last point here https://mattermost.atlassian.net/browse/MM-31083